src/univariate/affine.jl

```diff
@@ -0,0 +1,343 @@
+log_abs_det(x::AbstractMatrix) = first(logabsdet(x))
+log_abs_det(x::Real) = log(abs(x))
+_eps(x) = zero(x)
+_eps(x::AbstractFloat) = eps(x)
+
+"""
+    AffineDistribution(μ, σ, ρ)
+
+A shifted and scaled (affinely transformed) version of `ρ`.
+
+If ``Z`` is a random variable with distribution `ρ`, then `AffineDistribution` with
+parameters `μ`, `σ`, `ρ` is the distribution of the random variable
+```math
+X = μ + σ * Z
+```
+
+If `ρ` is a discrete distribution, the probability mass function of the transformed 
+distribution is given by
+```math
+P(X = x) = P\\left(Z = \\frac{x-μ}{σ} \\right).
+```
+
+If `ρ` is a continuous distribution, the probability density function of
+the transformed distribution is given by
+```math
+f(x) = \\frac{1}{σ} ρ \\! \\left( \\frac{x-μ}{σ} \\right).
+```
+
+If `ρ` falls under a location-scale family, `affine` will attempt to return a 
+transformed distribution of type `ρ` if an `affine` method has been implemented. 
+Otherwise, it will fall back on the `AffineDistribution` wrapper type.
+
+```julia
+d = σ * ρ + μ       # Create location-scale transformed distribution
+params(d)           # Get the parameters, i.e. (μ, scale, ρ)
+location(d)         # Get the location parameter
+scale(d)            # Get the scale parameter
+```
+"""
+struct AffineDistribution{
+        F<:VariateForm,
+        S<:ValueSupport,
+        Tμ<:ArrayLike,
+        Tσ<:ArrayLike,
+        D<:Distribution{F,S}
+    } <: Distribution{F,S}
+    μ::Tμ
+    σ::Tσ
+    ρ::D
+end
+
+
+#### Aliases
+@deprecate LocationScale(args...; kwargs...) AffineDistribution(args...)
+const ContinuousAffine{F,Tμ,Tσ,D} = AffineDistribution{F,Continuous,Tμ,Tσ,D}
+const DiscreteAffine{F,Tμ,Tσ,D} = AffineDistribution{F,Discrete,Tμ,Tσ,D}
+
+const UnivariateAffine{S,Tμ,Tσ,D} = AffineDistribution{Univariate,S,Tμ,Tσ,D}
+const MultivariateAffine{S,Tμ,Tσ,D} = AffineDistribution{Multivariate,S,Tμ,Tσ,D}
+
+
+#### Constructors
+
+"""
+    +(μ, dist)
+
+Return a version of `dist` that has been translated by `μ`.
+"""
+function Base.:+(μ::Real, ρ::UnivariateDistribution)
+    return AffineDistribution(μ, one(μ), ρ)
+end
+
+function Base.:+(μ::AbstractArray{<:Real, N}, ρ::Distribution{ArrayLikeVariate{N}, S}) where {N, S}
+    if size(μ) ≠ size(dist) 
+        throw(DimensionMismatch("array and distribution have different sizes."))
+    end
+    return AffineDistribution(μ, Diagonal(Ones(μ)), ρ)
```

Comment:
It's possible, but `I` can be a pain to deal with because it's not an array. This can throw off code that assumes `scale(x)` returns an array, for instance.